### PR TITLE
Improve Dashboard Widgets and Email Address Widget (CO-2679)

### PR DESCRIPTION
**Pull request by spartanicus**
This PR:

    ...allows for widget headers and footers generally.
    ...limits the email addresses shown in the Email Address widget by type.
    ...provides a configuration to limit the number of emails allowed in the Email Address widget.
    ...disallows creation of duplicate email addresses within the Email Address widget.
    ...adds the feature "Retain Last Email Address" disabling the deletion of the last email address
    ...adds the feature "Allow Replacement" allowing the replacement of an email address (with validation)

This PR obviates PR-531 which was not yet merged at the time this PR was created.

### CODE REVIEW

File: app/AvailablePlugin/EmailAddressWidget/Model/EmailAddressWidgetVerification.php [hunk 1]
```diff
@@ -133,6 +138,54 @@ public function addEmailToPerson($token, $actorCoPersonId) {
...
+        $actorCoPersonId,
+        ActionEnum::EmailAddressVerified,
+        _txt('pl.emailaddresswidget.rs.mail.updated.verified',
+          array($rec['EmailAddressWidgetVerification']['email'],
+                $rec['EmailAddressWidgetVerification']['email_id'])));
+      
+      $this->delete($rec['EmailAddressWidgetVerification']['id']);
+      return $EmailAddress->id;
+      // Delete the Verification Request table record and return
+    } catch(Exception $e) {
```
Comment (ioigoume) [inline app/AvailablePlugin/EmailAddressWidget/Model/EmailAddressWidgetVerification.php:185]: probably you need to delete the verification on an Exception

**Review by ioigoume — COMMENTED** [review]
It looks good. I added a small comment. Tested the flow and it works as expected.

File: app/AvailablePlugin/EmailAddressWidget/Model/EmailAddressWidgetVerification.php (hunk 1 above)
Comment (spartanicus) [reply app/AvailablePlugin/EmailAddressWidget/Model/EmailAddressWidgetVerification.php]: Code updated to delete the EmailAddressWidgetVerification table row on this exception.

**Comment by spartanicus** [comment]
Rebased against the latest develop (at 4.3.0 release)

**Pull request merged**